File: account-service/src/main/java/com/wks/servicemarketplace/accountservice/core/utils/ValidationException.java
package com.wks.servicemarketplace.accountservice.core.utils;

import java.util.Collections;
import java.util.Map;
import java.util.stream.Collectors;

public class ValidationException extends RuntimeException {

    private final Map<String, String> fields;

    public ValidationException(Map<String, String> fields) {
        super(buildErrorMessage(fields));
        this.fields = Collections.unmodifiableMap(fields);
    }

    private static String buildErrorMessage(final Map<String, String> fields) {
        return fields.entrySet()
                .stream()
                .map(entry -> String.format("%s: %S", entry.getKey(), entry.getValue()))
                .collect(Collectors.joining(","));
    }

    public Map<String, String> getFields() {
        return fields;
    }
}

File: job-service/src/main/kotlin/com/wks/servicesmarketplace/jobservice/core/utils/ValidationException.kt
package com.wks.servicesmarketplace.jobservice.core.utils

import java.util.*

class ValidationException(fields: Map<String, String>) : RuntimeException(buildErrorMessage(fields)) {
    val fields: Map<String, String> = Collections.unmodifiableMap(fields)

    companion object {
        private fun buildErrorMessage(fields: Map<String, String>): String {
            return fields.entries
                    .map { "${it.key}: ${it.value}" }
                    .joinToString(",")
        }
    }

}